lib/Services/JsonWebTokenBuilderService.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace SimpleSAML\Module\oidc\Services;
+
+use DateTimeImmutable;
+use Exception;
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Encoding\ChainedFormatter;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Lcobucci\JWT\UnencryptedToken;
+use League\OAuth2\Server\Exception\OAuthServerException;
+use ReflectionException;
+use SimpleSAML\Module\oidc\Utils\FingerprintGenerator;
+use SimpleSAML\Module\oidc\Utils\UniqueIdentifierGenerator;
+
+class JsonWebTokenBuilderService
+{
+    protected ConfigurationService $configurationService;
+    protected Configuration $jwtConfig;
+
+    /**
+     * @throws ReflectionException
+     * @throws Exception
+     */
+    public function __construct(
+        ?ConfigurationService $configurationService = null
+    ) {
+        $this->configurationService = $configurationService ?? new ConfigurationService();
```

Comment:
I don't understand why ConfigurationService is optional. Isn't this instantiated in the Container and so the dependency should be available?

Reply:
Check comment.